src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -102,40 +180,55 @@
           [content.deleted/deleted-message message-data context]
           [message/message-with-reactions message-data context keyboard-shown])]))])
 
-(defn messages-list
-  [{:keys [chat-id] :as chat}]
-  (let [context        (rf/sub [:chats/current-chat-message-list-view-context])
-        messages       (rf/sub [:chats/raw-chat-messages-stream chat-id])
-        keyboard-shown (atom false)
-        bottom-space   15]
-    [rn/view
-     {:style {:flex 1}}
-     ;; NOTE: DO NOT use anonymous functions for handlers
-     [rn/flat-list
+(defn scroll-handler
+  [event initial-y scroll-y]
+  (let [content-size-y (- (oops/oget event "nativeEvent.contentSize.height")
+                          (oops/oget event "nativeEvent.layoutMeasurement.height"))
+        current-y      (+ (oops/oget event "nativeEvent.contentOffset.y") initial-y)]
+    (reanimated/set-shared-value scroll-y (- content-size-y current-y))))
+
+(defn messages-view
+  [{:keys [chat insets initial-y scroll-y cover-bg-color]}]
+  (let [render-data (rf/sub [:chats/current-chat-message-list-view-context])
+        messages    (rf/sub [:chats/raw-chat-messages-stream (:chat-id chat)])]
+    [rn/view {:style {:flex 1}}
+     [reanimated/flat-list
       {:key-fn                       list-key-fn
        :ref                          list-ref
-       :header                       [list-header chat]
-       :footer                       [list-footer chat]
+       :header                       (when (= (:chat-type chat) constants/private-group-chat-type)
+                                       [list-group-chat-header chat])
+       :footer                       [list-footer
+                                      {:chat           chat
+                                       :insets         insets
+                                       :scroll-y       scroll-y
+                                       :cover-bg-color cover-bg-color}]
```

Comment:
To avoid the top-part flickering when scrolling fast, you could try to find a way in which the `list-footer` won't be rendered unless the flat-list has rendered all items